pilot/pkg/config/kube/gateway/templates/deployment.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    {{ toYamlMap .Annotations | nindent 4 }}
+  labels:
+    {{ toYamlMap .Labels
+      (strdict "gateway.istio.io/managed" "istio.io-gateway-controller")
+      | nindent 4}}
+  name: {{.Name}}
+  namespace: {{.Namespace}}
+  ownerReferences:
+  - apiVersion: gateway.networking.k8s.io/v1alpha2
+    kind: Gateway
+    name: {{.Name}}
```

Comment:
Would be really nice to use the same template elements/names as helm, it is harder to read/understand otherwise. And if this gets more complicated - we may reuse the helm-applying code we use in istioctl.